src/cpp/core/Base64Tests.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * StringUtils.cpp
+ *
+ * Copyright (C) 2009-12 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <tests/TestThat.hpp>
+
+#include <core/Error.hpp>
+#include <core/Base64.hpp>
+#include <core/StringUtils.hpp>
+
+namespace rstudio {
+namespace core {
+namespace base64 {
+
+context("Base64 Encoding")
+{
+   std::string encoded;
+   test_that("Various small strings encode correctly")
+   {
+      encode("a", &encoded);
+      expect_true(encoded == "YQ==");
+      
+      encode("ab", &encoded);
+      expect_true(encoded == "YWI=");
+      
+      encode("abc", &encoded);
+      expect_true(encoded == "YWJj");
+      
+      encode("abcd", &encoded);
+      expect_true(encoded == "YWJjZA==");
+      
+      encode("abcde", &encoded);
+      expect_true(encoded == "YWJjZGU=");
+      
+      encode("abcdef", &encoded);
+      expect_true(encoded == "YWJjZGVm");
+   }
+   
+   std::string decoded;
+   test_that("Various small strings decode correctly")
+   {
+      decode("YQ==", &decoded);
+      expect_true(decoded == "a");
+      
+      decode("YWI=", &decoded);
+      expect_true(decoded ==  "ab");
+      
+      decode("YWJj", &decoded);
+      expect_true(decoded == "abc");
+      
+      decode("YWJjZA==", &decoded);
+      expect_true(decoded == "abcd");
+      
+      decode("YWJjZGU=", &decoded);
+      expect_true(decoded == "abcde");
+      
+      decode("YWJjZGVm", &decoded);
+      expect_true(decoded == "abcdef");
+   }
+   
+   test_that("Contents are preserved in encode / decode process")
```

Comment:
(although you may want to set the random seed so you don't get a failure you can't reproduce)